runtime/gc_structs/ClassIteratorClassSlots.cpp

```diff
@@ -66,9 +67,15 @@ GC_ClassIteratorClassSlots::nextSlot()
 		_state += 1;
 
 	case classiteratorclassslots_state_array_class_slots:
-		slotPtr = _classArrayClassSlotIterator.nextSlot();
-		if(NULL != slotPtr) {
-			return slotPtr;
+		/*
+		 * Checking sharedITable is an optimization that only checks booleanArrayClass Interfaces
+		 * since all array claseses share the same ITable.
+		 */
+		if (!GC_ClassModel::usesSharedITable(_javaVM, _clazzPtr)) {
```

Comment:
I believe we can optimize this - instead to call function for every single element we can save result in `bool _shouldScanInterfaces` in constructor. And we don't need to keep `_javaVM` in this case